source/Wormhole.cpp

```diff
@@ -97,6 +98,14 @@ void Wormhole::Load(const DataNode &node)
 			else
 				child.PrintTrace("Missing value for attribute:");
 		}
+		else if(key == "color" && hasValue)
+		{
+			if(child.Size() >= 3 + valueIndex)
+				linkColor = ExclusiveItem<Color>(Color(child.Value(valueIndex),
+						child.Value(valueIndex + 1), child.Value(valueIndex + 2)));
+			else if(child.Size() >= 1 + valueIndex)
+				linkColor = ExclusiveItem<Color>(GameData::Colors().Get(child.Token(valueIndex)));
+		}
```

Comment:
There should either be an error message in the case of a malformed `color` node or these nested `if` statements should be moved up a level.
```suggestion
		else if(key == "color" && hasValue)
		{
			if(child.Size() >= 3 + valueIndex)
				linkColor = ExclusiveItem<Color>(Color(child.Value(valueIndex),
						child.Value(valueIndex + 1), child.Value(valueIndex + 2)));
			else if(child.Size() >= 1 + valueIndex)
				linkColor = ExclusiveItem<Color>(GameData::Colors().Get(child.Token(valueIndex)));
			else
				child.PrintTrace("Error: malformed 'color' node.");
		}
```
```suggestion
		else if(key == "color" && hasValue && child.Size() >= 3 + valueIndex)
			linkColor = ExclusiveItem<Color>(Color(child.Value(valueIndex),
					child.Value(valueIndex + 1), child.Value(valueIndex + 2)));
		else if(key == "color" && hasValue && child.Size() >= 1 + valueIndex)
			linkColor = ExclusiveItem<Color>(GameData::Colors().Get(child.Token(valueIndex)));
```